sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cas-integration/_tests_/cas-invoices-batches-creation.scheduler.e2e-spec.ts

```diff
@@ -280,5 +283,202 @@ describe(
+          "Checking for pending receipts.",
+          "Found 1 pending receipts.",
+          `Creating invoice for receipt ID ${provincialDisbursementReceiptWithoutInvoice.id}.`,
+          `Invoice SIMS-INVOICE-1-${casSupplier.supplierNumber} created for receipt ID ${provincialDisbursementReceiptWithoutInvoice.id}.`,
+          `Created invoice detail for award BCAG(CR).`,
+          `Created invoice detail for award BCAG(DR).`,
+          `CAS invoices batches creation process executed.`,
+        ]),
+      ).toBe(true);
+      expect(
+        mockedJob.containLogMessages([
+          `Creating invoice for receipt ID ${provincialDisbursementReceiptWithInvoice.id}.`,
+        ]),
+      ).toBe(false);
+    });
+
+    it("Should interrupt the process when an invoice is trying to be generated but there are no distribution accounts available to create the invoice details.", async () => {
+      // Arrange
+      const BC_GRANT_WITHOUT_DISTRIBUTION_ACCOUNT = "BCAG";
+      const casSupplier = await saveFakeCASSupplier(db, undefined, {
+        initialValues: {
+          supplierStatus: SupplierStatus.VerifiedManually,
+        },
+      });
+      const student = await saveFakeStudent(db.dataSource, { casSupplier });
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+          disbursementValues: [
+            createFakeDisbursementValue(
+              DisbursementValueType.BCGrant,
+              BC_GRANT_WITHOUT_DISTRIBUTION_ACCOUNT,
+              351,
+              { effectiveAmount: 350 },
+            ),
+            createFakeDisbursementValue(
+              DisbursementValueType.BCTotalGrant,
+              "BCSG",
+              351,
+              { effectiveAmount: 350 },
+            ),
+          ],
+        },
+      );
+      const [firstDisbursementSchedule] =
+        application.currentAssessment.disbursementSchedules;
+      await saveFakeDisbursementReceiptsFromDisbursementSchedule(
+        db,
+        firstDisbursementSchedule,
+      );
+      // Change the distribution account to be inactive.
+      await db.casDistributionAccount.update(
+        {
+          awardValueCode: BC_GRANT_WITHOUT_DISTRIBUTION_ACCOUNT,
+          offeringIntensity: OfferingIntensity.fullTime,
+        },
+        { isActive: false },
+      );
+
+      // Queued job.
+      const mockedJob = mockBullJob<void>();
+
+      // Act/Assert
+      await expect(processor.processQueue(mockedJob.job)).rejects.toThrow(
+        `No distribution accounts found for award ${BC_GRANT_WITHOUT_DISTRIBUTION_ACCOUNT} and offering intensity ${OfferingIntensity.fullTime}.`,
+      );
+
+      // Revert back the distribution account to be active.
+      await db.casDistributionAccount.update(
+        {
+          awardValueCode: BC_GRANT_WITHOUT_DISTRIBUTION_ACCOUNT,
+          offeringIntensity: OfferingIntensity.fullTime,
+        },
+        { isActive: true },
+      );
```

Comment:
I would suggest to use a particular distribution account that is for invalid scenario instead. The benefit is that other tests potentially does not require this one to be executed completely.

Reply:
I will change to an invalid BC grant, like `ABCD`, but I did not follow how this test would interfere in any way with other tests.

Reply:
Changed to `ABCD`.